code/game/machinery/doors/blast_door.dm

```diff
@@ -107,40 +107,44 @@
 // Description: Opens the door. No checks are done inside this proc.
 /obj/machinery/door/blast/proc/force_open()
 	set waitfor = FALSE
-	operating = 1
+	operating = TRUE
 	playsound(src.loc, open_sound, 100, 1)
 	flick(icon_state_opening, src)
-	set_density(0)
+	addtimer(CALLBACK(src, .proc/open_final), 0.6 SECONDS, TIMER_UNIQUE)
+
+/obj/machinery/door/blast/open_final()
+	set_density(FALSE)
 	update_nearby_tiles()
 	update_icon()
-	set_opacity(0)
-	sleep(15)
+	set_opacity(FALSE)
 	layer = open_layer
-	operating = 0
+	operating = FALSE
 
 // Proc: force_close()
 // Parameters: None
 // Description: Closes the door. No checks are done inside this proc.
 /obj/machinery/door/blast/proc/force_close()
-	operating = 1
+	operating = TRUE
 	playsound(src.loc, close_sound, 100, 1)
 	layer = closed_layer
 	flick(icon_state_closing, src)
-	set_density(1)
+	addtimer(CALLBACK(src, .proc/close_final), 0.6 SECONDS, TIMER_UNIQUE)
+
+/obj/machinery/door/blast/close_final()
+	set_density(TRUE)
 	update_nearby_tiles()
 	update_icon()
-	set_opacity(1)
-	sleep(15)
-	operating = 0
+	set_opacity(TRUE)
+	operating = FALSE
 
 // Proc: force_toggle()
 // Parameters: None
 // Description: Opens or closes the door, depending on current state. No checks are done inside this proc.
 /obj/machinery/door/blast/proc/force_toggle(to_open = density)
 	if(to_open)
-		force_open()
+		open()
 	else
-		force_close()
+		close()
```

Comment:
Why this change? I believe this is used with radio stuff to remotely open groups of doors, so having them not force can leave them in weird staggered states.

EDIT: on reviewing the rest I can sort of see why you did this, but I'd suggest either not doing it (assuming it was just wrong before) or renaming the proc.